### ISSUE TITLE
ansible-vault wont open decrypt file with env vars set with multiple keys

### ISSUE THREAD

**Issue by orderedchaosdev** [ansible/ansible#38146]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ansible-vault

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  python version = 2.7.14 (default, Mar 21 2018, 16:45:24) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

##### CONFIGURATION

```
DEFAULT_VAULT_IDENTITY_LIST(env: ANSIBLE_VAULT_IDENTITY_LIST) = ['default@~/.ssh/vault/default.pass', 'bench@~/.ssh/vault/bench.pass']
```

##### SUMMARY

In ansible 2.4 with those environmental vars set I can edit any vaulted file that was encrypted with either of the vault keys. 
In ansible 2.5 it won't open the file. 
##### STEPS TO REPRODUCE

Create your file to be vaulted.
Set environment vars as I did with vault password file paths and ids
encrypt a file with the default key.
Then try ansible-vault edit file_name
It will throw an error.

```
Warning due to bug in ansible 2.4 to encrypt you need to do
unset ANSIBLE_VAULT_INDENTITY_LIST
ansible-vault encrypt ./all_vault.yml --vault-id default@~/.ssh/vault/default.pass
export ANSIBLE_VAULT_IDENTITY_LIST='default@~/.ssh/vault/default.pass, bench@~/.ssh/vault/bench.pass'
ansible-vault edit ./all_vault.yml
In ansible 2.5
export ANSIBLE_VAULT_IDENTITY_LIST='default@~/.ssh/vault/default.pass, bench@~/.ssh/vault/bench.pass'
ansible-vault encrypt --encrypt-vault-id default ./all_vault.yml
ansible-vault edit ./all_vault.yml
```

##### EXPECTED RESULTS

I expected the file to open like it would in ansible 2.4 steps.

##### ACTUAL RESULTS

It throws the error below which is also wrong since for edit --encrypt-vault-id is not a valid parameter.

```
ansible-vault edit ./all_vault.yml
Usage: ansible-vault edit [options] file_name

encryption/decryption utility for Ansible data files

Options:
  --ask-vault-pass      ask for vault password
  -h, --help            show this help message and exit
  --new-vault-id=NEW_VAULT_ID
                        the new vault identity to use for rekey
  --new-vault-password-file=NEW_VAULT_PASSWORD_FILE
                        new vault password file for rekey
  --vault-id=VAULT_IDS  the vault identity to use
  --vault-password-file=VAULT_PASSWORD_FILES
                        vault password file
  -v, --verbose         verbose mode (-vvv for more, -vvvv to enable
                        connection debugging)
  --version             show program's version number and exit

 See 'ansible-vault <command> --help' for more information on a specific
command.
ERROR! The vault-ids default,bench are available to encrypt. Specify the vault-id to encrypt with --encrypt-vault-id
```

**Comment by orderedchaosdev** [comment]
Also I tried encrypting the vault file with both the default which makes it ```$ANSIBLE_VAULT;1.1;AES256```
And bench which makes it ```$ANSIBLE_VAULT;1.2;AES256;bench``` both cases have the same issue

**Comment by alikins** [comment]
I think this should be resolved by https://github.com/ansible/ansible/pull/35923  
(merge in devel as 6e737c8cb66df1500dba1c74369314dd8f65867c, there is
a backport for stable-2.5 https://github.com/ansible/ansible/pull/38003 intended
for 2.5.1)

Related: https://github.com/ansible/ansible/issues/35834

**Comment by AquaL1te** [comment]
This is ridicules, do the Ansible devs even use CI themselves to test their code before a release? This is the second time, this also happened with Ansible 2.3 or 2.4... Why is every new point release so poorly tested?... Useless...

**Comment by bcoca** [comment]
@AquaL1te , curious since this feature was added in 2.4 .. hard  for it to 'regress' in any previous version

**Comment by ericdost** [comment]
I believe this is the same bug that I was hitting.  If `vault_password_file` is specified in `ansible.cfg`, You get an error trying to encrypt a string with a specific `vault-password-file`.

Here is the error: `ERROR! The vault-ids default,default are available to encrypt. Specify the vault-id to encrypt with --encrypt-vault-id`

Comment out `vault_password_file` and then it generates the string.

Here is my Ansible version:

```
$ ansible --version
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/ansible/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

**Comment by Zauxst** [comment]
I'm on 2.5.4 ; What @ericdost suggested has worked. Removed the vault_password_file and it functions.

**Comment by sivel** [comment]
I can confirm this was resolved in 6e737c8cb66df1500dba1c74369314dd8f65867c
